ocaml/xe-cli/newcli.ml

```diff
@@ -358,6 +358,11 @@ let main_loop ifd ofd permitted_filenames =
           marshal ofd (Blob End);
           Unix.close fd
         with
+        | Unix.Unix_error(Unix.EPIPE,fn,arg) as e ->
+          if length > Constants.max_cli_upload_bytes then
```

Comment:
if we exceed the upload limit, don't we want to fail before trying to read? (i.e. should we move this check before the try block?)

Reply:
The server-side (xapi) uses `Constants.max_cli_upload_bytes` to implement the size constraint. This code is for the client (xe), which may be older or newer than the server. Using the same `Constants.max_cli_upload_bytes` is a guess at best, because future versions of xapi may be stricter or less strict (older ones didn't have the constraint), while the same xe binary can be used to communicate with them.

It would therefore be better to not rely on `Constants.max_cli_upload_bytes` here. It would be better if xapi can tell xe why it is going to hang up, but to keep it simple, I would drop the if statement here and just report that the file upload failed. E.g. just `"Failed to upload file %s"`.